interpreter/convert.go

```diff
@@ -18,39 +22,44 @@ func convertToStarlark(value interface{}) (starlark.Value, error) {
 	case uint64:
 		return starlark.MakeUint64(v), nil
 	case float32:
+		return starlark.Float(v), nil
 	case float64:
 		return starlark.Float(v), nil
 	case string:
 		return starlark.String(v), nil
-	case []interface{}:
-		var elems []starlark.Value
-		for _, elem := range v {
-			val, err := convertToStarlark(elem)
-			if err != nil {
-				return nil, err
-			}
-			elems = append(elems, val)
-		}
-		return starlark.NewList(elems), nil
-	case map[interface{}]interface{}:
-		dict := starlark.NewDict(len(v))
-
-		for index, elem := range v {
-			key, err := convertToStarlark(index)
-			if err != nil {
-				return nil, err
+	default:
+		reflectV := reflect.ValueOf(value)
+		switch reflectV.Kind() {
+		case reflect.Slice:
+			var elems []starlark.Value
+			for i := 0; i < reflectV.Len(); i++ {
+				val, err := convertToStarlark(reflectV.Index(i).Interface())
+				if err != nil {
+					return nil, err
+				}
+				elems = append(elems, val)
 			}
+			return starlark.NewList(elems), nil
+		case reflect.Map:
+			dict := starlark.NewDict(len(reflectV.MapKeys()))
 
-			val, err := convertToStarlark(elem)
-			if err != nil {
-				return nil, err
-			}
+			iter := reflectV.MapRange()
+			for iter.Next() {
+				key, err := convertToStarlark(iter.Key().Interface())
+				if err != nil {
+					return nil, err
```

Comment:
nit: Wrap the error here to include the key of the element that failed to convert.

Reply:
key is a Value here, I have no idea how to force this to happen. I added a wrap, but it may be insufficient, we will have to see if this error ever comes up and fix it then /shrug